verify.go

```diff
@@ -19,19 +19,41 @@ const (
 	issuerGoogleAccountsNoScheme = "accounts.google.com"
 )
 
-// keySet is an interface that lets us stub out verification policies for
-// testing. Outside of testing, it's always backed by a remoteKeySet.
-type keySet interface {
-	verify(ctx context.Context, jws *jose.JSONWebSignature) ([]byte, error)
+// KeySet is a set of publc JSON web keys that can be used to validate the signature
+// of JSON web tokens. This is expected to be backed by a remote key set through
+// provider metadata discovery or an in-memory set of keys delivered out-of-band.
+type KeySet interface {
+	// VerifySignature parses the JSON web token, verifies the signature, and returns
+	// the raw payload. Header and claim fields are validated by other parts of the
+	// package. The KeySet does not need to check values such as signature algorithm,
```

Comment:
is there a reason you mentioned this specifically? are the rest of the values checked somewhere else?

Reply:
The IDTokenVerifier already performs these checks. Going to update the docstring:

```
        // VerifySignature parses the JSON web token, verifies the signature, and returns                              
        // the raw payload. Header and claim fields are validated by other parts of the
        // package. For example, the KeySet does not need to check values such as signature 
        // algorithm, issuer, and audience since the IDTokenVerifier validates these values
        // independently.                                  
```

Does that work?